lib/temporal/connection/grpc.rb

```diff
@@ -31,14 +31,15 @@ def initialize(host, port, identity)
         @poll_request = nil
       end
 
-      def register_namespace(name:, description: nil, global: false, retention_period: 10)
+      def register_namespace(name:, description: nil, global: false, retention_period: 10, namespace_data: nil)
```

Comment:
Code seems fine, but I don't see why `retention_period` has to be "days".

Reply:
I guess legacy reasons, happy to update this but don't want to break any existing functionality if we do

Reply:
Based on your other response that this is how long workflow events are retained, then I withdraw my comment. I thought it was namespace expiration.